src/clp_ffi_py/ir/Query.hpp

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * Constructs a new query object with the given timestamp range with an
+     * empty wildcard list.
+     * @param search_time_lower_bound Start of search time range (inclusive).
+     * @param search_time_upper_bound End of search time range (inclusive).
+     * @param search_termination_margin The margin used to determine the search
+     * termination timestamp.
+     */
+    explicit Query(
+            ffi::epoch_time_ms_t search_time_lower_bound,
+            ffi::epoch_time_ms_t search_time_upper_bound,
+            ffi::epoch_time_ms_t search_termination_margin = cDefaultSearchTerminationMargin
+    )
+            : m_lower_bound_ts{search_time_lower_bound},
+              m_upper_bound_ts{search_time_upper_bound},
+              m_search_termination_ts{
+                      (cTimestampMax - search_termination_margin > search_time_upper_bound)
+                              ? search_time_upper_bound + search_termination_margin
+                              : cTimestampMax} {
+        throw_if_ts_range_invalid();
+    }
+
+    /**
+     * Constructs a new query object with the given timestamp range and a
+     * wildcard list.
+     * @param search_time_lower_bound Start of search time range (inclusive).
+     * @param search_time_upper_bound End of search time range (inclusive).
+     * @param wildcard_list A reference to the wildcard queries vector, whose
+     * data will be transferred using std::move to initialize m_wildcard_list.
+     * @param search_termination_margin The margin used to determine the search
+     * termination timestamp. By default, this value is set to
+     * cDefaultSearchTerminationMargin.
+     */
+    explicit Query(
+            ffi::epoch_time_ms_t search_time_lower_bound,
+            ffi::epoch_time_ms_t search_time_upper_bound,
+            std::vector<WildcardQuery> wildcard_list,
+            ffi::epoch_time_ms_t search_termination_margin = cDefaultSearchTerminationMargin
+    )
+            : m_lower_bound_ts{search_time_lower_bound},
+              m_upper_bound_ts{search_time_upper_bound},
+              m_search_termination_ts{
+                      (cTimestampMax - search_termination_margin > search_time_upper_bound)
+                              ? search_time_upper_bound + search_termination_margin
+                              : cTimestampMax},
+              m_wildcard_list{std::move(wildcard_list)} {
+        throw_if_ts_range_invalid();
+    }
+
+    [[nodiscard]] auto get_lower_bound_ts() const -> ffi::epoch_time_ms_t {
+        return m_lower_bound_ts;
+    }
+
+    [[nodiscard]] auto get_ts_upper_bound_ts() const -> ffi::epoch_time_ms_t {
+        return m_upper_bound_ts;
+    }
+
+    /**
+     * @param ts Input timestamp.
+     * @return true if the given timestamp is in the search time range bounded
+     * by the lower bound and the upper bound timestamp (inclusive).
+     */
+    [[nodiscard]] auto ts_in_range(ffi::epoch_time_ms_t ts) const -> bool {
```

Comment:
For the three match methods, how about:
* `matches_time_range`
* `matches_wildcard_queries`
* `matches`